### ISSUE TITLE
`axolotl train` Rejects HTTPS Config URLs That the CLI Docs Say Are Supported

### ISSUE THREAD

**Issue by davanstrien** [axolotl-ai-cloud/axolotl#4041]
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

`docs/cli.qmd` says "The config file can be local or a URL to a raw YAML file", so `axolotl train https://.../config.yml` should download the config and train.

### Current behaviour

Click rejects the URL before Axolotl's code runs:

```
Usage: axolotl train [OPTIONS] CONFIG
Try 'axolotl train --help' for help.

Error: Invalid value for 'CONFIG': Path 'https://raw.githubusercontent.com/axolotl-ai-cloud/axolotl/main/examples/llama-3/lora-1b.yml' does not exist.
```

### Steps to reproduce

In the `axolotlai/axolotl:main-latest` image (reports 0.20.0.dev0; the 0.19.0 image behaves the same). No GPU needed:

```bash
axolotl train https://raw.githubusercontent.com/axolotl-ai-cloud/axolotl/main/examples/llama-3/lora-1b.yml
```

### Config yaml

```yaml

```

### Possible solution

The `config` argument of `train` (and the other commands) is `click.Path(exists=True, path_type=str)` in `src/axolotl/cli/main.py`, since the Click CLI in #2107. Click rejects the URL before `load_cfg` reaches `check_remote_config` (added in #1277).

Options: accept strings that start with `https://` before the exists check, or drop `exists=True` and let `load_cfg` raise for missing local files.

Not tested: `python -m axolotl.cli.train <url>` may still work, since that entry point goes through `fire`.

### Which Operating Systems are you using?

- [x] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.12

### axolotl branch-commit

main (855bcac); main-latest image 0.20.0.dev0; also 0.19.0

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by mira687** [comment]
Confirmed the cause you found, and traced what happens on the other side of it, because it changes
whether that fix is enough.

Nothing between Click and `load_cfg` opens the config on the plain path. With no `--sweep`,
`generate_config_files` yields the string unchanged (`cli/utils/train.py`, `if not sweep: yield
config, False`), and `launch_training` appends it to `["accelerate","launch",...,"-m","axolotl.cli.train"]`
— the fire entry point — so `do_cli` → `load_cfg` → `check_remote_config` downloads it in the child.
`--launcher python` calls `do_cli` in-process and lands in the same place. So relaxing the
`exists=True` check is genuinely sufficient for `axolotl train <url>`.

Your untested fallback works, and it is what `axolotl train` already shells out to. I couldn't
install axolotl here, so I only tested the parsing, against the identical signature (fire 0.7.1,
py3.12): the URL arrives as `str`, not coerced to `Path` by the `Union[Path, str]` annotation, so
`check_remote_config`'s `isinstance(config, str) and config.startswith("https://")` gate passes, and
`--num-epochs=2`-style options (what `build_command` emits) arrive as `{'num_epochs': 2}`.

Two branches the relaxation alone would not cover, because both read the config in-process and never
reach `check_remote_config`:

- `--sweep`: `generate_config_files` does `open(config)` at `cli/utils/train.py:86` to build the
  permutations.
- `--cloud`: `do_cli_train` does `open(config).read()` at `cli/cloud/__init__.py:49` to ship the YAML
  text (same at `:68` for `lm_eval`).

Both would raise a bare `FileNotFoundError` on a URL once Click stops rejecting it.

If that matters, a custom `click.ParamType` whose `convert()` runs `check_remote_config` and returns
the local path would cover all three branches at once and keep a clean usage error for missing local
files. The same `click.Path(exists=True, path_type=str)` is on nine `config` arguments (preprocess,
train, evaluate, inference, merge-sharded-fsdp-weights, merge-lora, vllm-serve, quantize, export),
and `docs/cli.qmd:24` makes the URL claim for `axolotl <command> [config.yml]` generally.
